### PR TITLE
Wasm

### PR DESCRIPTION
**Pull request by rdaly525**
Should merge rm-family before this branch.

NYI
-ctz,clz,popcnt
-tests for all the instructions

### CODE REVIEW

File: examples/wasm/ir.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+#Conversion ops
+
+WASM.add_peak_instruction("i32_wrap", Input64,Output32, lambda x : x[:32])
+WASM.add_peak_instruction("i64_extend_s", Input32,Output64, lambda x: x.sext(32))
+WASM.add_peak_instruction("i64_extend_u", Input32,Output64, lambda x: x.zext(32))
+
+
+for name, inst in WASM.instructions.items():
+    print(name,inst)
```
Comment (cdonovick) [inline examples/wasm/ir.py]: remove

File: tests/test_wasm.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    assert rotl_class.get_inputs().field_dict == dict(in0=Data,in1=Data)
+    assert rotl_class.get_outputs().field_dict == dict(out=Data)
+
+    #directed test
+
+    in0 = Data(0x12345678)
+    in1 = Data(8)
+    out = Data(0x34567812)
+    assert rotl_class()(in0,in1) == out
+
```
Comment (cdonovick) [inline tests/test_wasm.py]: needs more tests

Comment (rdaly525) [reply tests/test_wasm.py]: Agreed. I dont have bandwidth right now to add them. Id rather get core mapper stuff working.

**Comment by rdaly525** [comment]
Test issue #82

File: examples/wasm/ir.py (hunk 1 above)
Comment (rdaly525) [reply examples/wasm/ir.py]: removed

**Pull request merged**